commit 7a51c88c35ae6fbd858a50eef3f30719242659c5
Author: Neil Armstrong <narmstrong@baylibre.com>
Date:   Wed Oct 11 17:23:12 2017 +0200

    ARM64: dts: meson-gxl: Add alternate ARM Trusted Firmware reserved memory zone
    
    commit 4ee8e51b9edfe7845a094690a365c844e5a35b4b upstream.
    
    This year, Amlogic updated the ARM Trusted Firmware reserved memory mapping
    for Meson GXL SoCs and products sold since May 2017 uses this alternate
    reserved memory mapping.
    But products had been sold using the previous mapping.
    
    This issue has been explained in [1] and a dynamic solution is yet to be
    found to avoid loosing another 3Mbytes of reservable memory.
    
    In the meantime, this patch adds this alternate memory zone only for
    the GXL and GXM SoCs since GXBB based new products stopped earlier.
    
    [1] http://lists.infradead.org/pipermail/linux-amlogic/2017-October/004860.html
    
    Fixes: bba8e3f42736 ("ARM64: dts: meson-gx: Add firmware reserved memory zones")
    Reported-by: Jerome Brunet <jbrunet@baylibre.com>
    Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>
    Signed-off-by: Kevin Hilman <khilman@baylibre.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/arch/arm64/boot/dts/amlogic/meson-gxl.dtsi b/arch/arm64/boot/dts/amlogic/meson-gxl.dtsi
index d8dd3298b15c..fb8d76a17bc5 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxl.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl.dtsi
@@ -49,6 +49,14 @@
 
 / {
 	compatible = "amlogic,meson-gxl";
+
+	reserved-memory {
+		/* Alternate 3 MiB reserved for ARM Trusted Firmware (BL31) */
+		secmon_reserved_alt: secmon@05000000 {
+			reg = <0x0 0x05000000 0x0 0x300000>;
+			no-map;
+		};
+	};
 };
 
 &ethmac {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1179_7a51c88c35ae6fbd858a50eef3f30719242659c5.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1179_7a51c88c35ae6fbd858a50eef3f30719242659c5.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
